graylog2-server/src/main/java/org/graylog2/security/AccessTokenImpl.java

```diff
@@ -18,15 +18,18 @@
 
 import com.google.common.collect.Maps;
 import org.bson.types.ObjectId;
-import org.graylog2.database.CollectionName;
+import org.graylog2.database.DbEntity;
 import org.graylog2.database.PersistedImpl;
 import org.graylog2.database.validators.FilledStringValidator;
 import org.graylog2.plugin.database.validators.Validator;
 import org.joda.time.DateTime;
 
 import java.util.Map;
 
-@CollectionName(AccessTokenImpl.COLLECTION_NAME)
+import static org.graylog2.security.AccessTokenImpl.COLLECTION_NAME;
+import static org.graylog2.security.AccessTokenImpl.NAME;
+
+@DbEntity(collection = COLLECTION_NAME, titleField = NAME)
```

Comment:
I've changed the approach so that by default you are forbidden to work with the entity (before "missing" readPermission meant "you can access").
It should solve the case.